libs/datasets/timeseries.py

```diff
@@ -610,6 +613,51 @@ def to_csv(self, path: pathlib.Path):
             provenance_path = str(path).replace(".csv", "-provenance.csv")
             self.provenance.sort_index().to_csv(provenance_path)
 
+    def join_columns(self, other: "MultiRegionTimeseriesDataset") -> "MultiRegionTimeseriesDataset":
+        """Joins the timeseries columns in `other` with those in `self`."""
+        if not other.latest_data.empty:
+            raise NotImplementedError("No support for joining other with latest_data")
+        other_df = other.data_with_fips.set_index([CommonFields.LOCATION_ID, CommonFields.DATE])
+        self_df = self.data_with_fips.set_index([CommonFields.LOCATION_ID, CommonFields.DATE])
+        other_geo_columns = set(other_df.columns) & set(GEO_DATA_COLUMNS)
+        other_ts_columns = (
+            set(other_df.columns) - set(GEO_DATA_COLUMNS) - set(TimeseriesDataset.INDEX_FIELDS)
+        )
+        common_ts_columns = other_ts_columns & set(self.data_with_fips.columns)
+        if common_ts_columns:
+            # columns to be joined need to be disjoint
+            raise ValueError(f"Columns are in both dataset: {common_ts_columns}")
+        common_geo_columns = list(set(self.data_with_fips.columns) & other_geo_columns)
+        # TODO(tom): fix geo columns check, no later than when self.data is changed to contain only
+        # timeseries
+        # self_common_geo_columns = self_df.loc[:, common_geo_columns].fillna("")
+        # other_common_geo_columns = other_df.loc[:, common_geo_columns].fillna("")
+        # try:
+        #    if (self_common_geo_columns != other_common_geo_columns).any(axis=None):
+        #        unequal_rows = (self_common_geo_columns != other_common_geo_columns).any(axis=1)
+        #        _log.info(
+        #            "Geo data unexpectedly varies",
+        #            self_rows=self_df.loc[unequal_rows, common_geo_columns],
+        #            other_rows=other_df.loc[unequal_rows, common_geo_columns],
+        #        )
+        #        raise ValueError("Geo data unexpectedly varies")
+        # except Exception:
+        #    _log.exception(f"Comparing df {self_common_geo_columns} to {other_common_geo_columns}")
+        #    raise
+        combined_df = pd.concat([self_df, other_df[list(other_ts_columns)]], axis=1)
```

Comment:
How does this behave with different index values? 

for instance differing dates in the two data frames. or a region in one df but not the other

Reply:
concat uses the index to do the right thing, unless ignore_index=True.